code/scripting/api/objs/intelentry.cpp

```diff
@@ -65,16 +65,14 @@ ADE_VIRTVAR(Description, l_Intelentry, "string", "Intel entry description", "str
 	if(!ade_get_args(L, "o|s", l_Intelentry.Get(&idx), &s))
 		return ade_set_error(L, "s", "");
 
-	if(idx < 0 || idx >= Intel_info_size)
+	if(idx < 0 || idx >= intel_info_size())
 		return ade_set_error(L, "s", "");
 
 	if(ADE_SETTING_VAR) {
 		if(s != nullptr) {
-			auto len = sizeof(Intel_info[idx].desc);
-			strncpy(Intel_info[idx].desc, s, len);
-			Intel_info[idx].desc[len - 1] = 0;
+			Intel_info[idx].desc = s;
 		} else {
-			strcpy_s(Intel_info[idx].desc, "");
+			Intel_info[idx].desc = "";
```

Comment:
You could also use `.clear()` here.

Reply:
Ok.

Reply:
Good to know.  I like the empty string though because it's explicit.